conda/recipes/meta.yaml

```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-{% set name = "datafusion" %}
+{% set name = "datafusion-python" %}
```

Comment:
Noticed that we're already publishing arrow-datafusion to conda-forge under the name `datafusion`:

- https://anaconda.org/conda-forge/datafusion

This shouldn't have any impact on the name of the package that actually gets imported, just prevents collision with the other package; does this name seem okay? cc @andygrove